File: sportsbook-app/payment-contracts/src/main/java/com/sportsbook/payment/contracts/saga/ReserveFundsCommand.java
package com.sportsbook.payment.contracts.saga;

import java.math.BigDecimal;

public record ReserveFundsCommand(
        String sagaId,
        String depositId,
        String playerId,
        BigDecimal amount,
        String currency,
        String paymentMethod,
        String instrumentId
) {}

File: sportsbook-app/payment-contracts/src/main/kotlin/com/sportsbook/payment/contracts/saga/ReserveFundsCommand.kt
package com.sportsbook.payment.contracts.saga

import java.math.BigDecimal

data class ReserveFundsCommand(
    val sagaId: String,
    val depositId: String,
    val playerId: String,
    val amount: BigDecimal,
    val currency: String,
    val paymentMethod: String,
    val instrumentId: String,
)
